core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -907,15 +910,20 @@ class SnappySession(@transient private val sc: SparkContext,
           case _ => // ignore
         }
         cacheManager.uncacheQuery(Dataset.ofRows(this, plan))
-        sessionCatalog.unregisterDataSourceTable(tableIdent, Some(br))
+        if(sessionCatalog.isTemporaryTable(tableIdent)){ // This is due to temp table
```

Comment:
So "table not found" kind of error was coming when trying to drop temp tables which was directly trying unregisterDataSourceTable?